src/utils/convertTimeStringToObj.ts

```diff
@@ -1,36 +1,50 @@
 import { TimeObjectSchema } from 'serina.schema';
-import { MERIDIEM, PM } from 'filters/time/time.constants';
+import TIME from 'filters/time/time.constants';
 import contains from './contains';
-import { remove, isValidTime } from 'utils';
+import { matchPattern } from 'utils';
 
-function convertTimeStringToObj(time: string): TimeObjectSchema {
+function getValidMatch(text: string, pattern: string): string {
+    const matched = matchPattern(text, pattern, false);
+    if (!matched || matched.length === 0) return null;
+    const [ value ] = matched;
+    return value;
+}
+
+function convertTimeStringToObj(timeString: string): TimeObjectSchema {
     let hour;
     let minute;
-
-    if (contains(time, `${MERIDIEM}`, false)) {
-        // 12H
-        const timeWithMeridiem = remove(time, `${MERIDIEM}`, false);
-        const [ hourStr, minStr ] = timeWithMeridiem.split(':');
-        hour = parseInt(hourStr, 10);
-        minute = parseInt(minStr, 10);
-
-        if (isNaN(minute)) minute = 0;
-
-        if (contains(time, `${PM}`,false)) {
-            // Post Meridiem (PM)
-            if (hour < 12) hour += 12;
-        } else {
-            // Ante Meridiem (AM)
-            if (hour === 12) hour = 0;
+    
+    const isRelativeTime = contains(timeString, TIME.RELATIVE_TIME_FILLER_WORDS);
+    
+    if (isRelativeTime) {
+        if (contains(timeString, TIME.VERBAL_QUANTIFIERS)) { // half or quarter
+            hour = getValidMatch(timeString, `(?<=${TIME.RELATIVE_TIME_FILLER_WORDS}( ))${TIME.HOUR_PART}`)
```

Comment:
I wanted to use ?<= operator, but it is not currently well supported. Might want to reconsider using it.

Reply:
Yeah you are right it is not well supported. But I'd say let's clean this up later, I'm thinking of improving this further after all the features have been implemented anyway

Looks like it is not even supported by Firefox
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp